projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java

```diff
@@ -40,8 +43,13 @@ public class VaultJobSecretsService implements com.vmware.taurus.secrets.service
   private final ObjectMapper objectMapper = new ObjectMapper();
   private final VaultOperations vaultOperations;
 
+  private final ConcurrentHashMap<String, VaultTeamCredentials> teamIdToCredentialsCache;
```

Comment:
Probablly you can have multiple instances of cache due to the Spring dependencies management.